module_utils/network_lsr/argument_validator.py

```diff
@@ -633,6 +704,177 @@ def _validate_post(self, value, name, result):
+                ),
+                ArgValidatorRange("dport", val_min=0, val_max=65534),
+                ArgValidatorStr(
+                    "family",
+                    default_value="ipv4",
+                    enum_values=["ipv4", "ipv6"],
+                ),
+                ArgValidatorIP(
+                    "from", family=family, default_value=None, plain_address=False
+                ),
+                ArgValidatorNum("from_len", default_value=0, val_min=0),
+                ArgValidatorNum(
+                    "fwmark", default_value=None, val_min=0, val_max=0xFFFFFFFF
+                ),
+                ArgValidatorNum(
+                    "fwmask", default_value=None, val_min=0, val_max=0xFFFFFFFF
+                ),
+                ArgValidatorStr("iif", default_value=None),
+                ArgValidatorBool("invert", default_value=None),
+                ArgValidatorNum("ipproto", default_value=None, val_min=0, val_max=255),
+                ArgValidatorStr("oif", default_value=None),
+                ArgValidatorRange("sport", val_min=0, val_max=65534),
+                ArgValidatorNum(
+                    "suppress_prefixlength", default_value=None, val_min=-1
+                ),
+                ArgValidatorRouteTable("table"),
+                ArgValidatorIP(
+                    "to", family=family, default_value=None, plain_address=False
+                ),
+                ArgValidatorNum("to_len", default_value=0, val_min=0),
+                ArgValidatorNum("tos", default_value=0, val_max=255),
+                ArgValidatorRange("uid", val_min=0, val_max=0xFFFFFFFF),
+            ],
+            default_value=None,
+        )
+        self.family = family
+
+    def _validate_post(self, value, name, result):
+        result["family"] = (socket.AF_INET, socket.AF_INET6)[result["family"] == "ipv6"]
+        if result["from"]:
+            network_from = result["from"]
+            family = network_from["family"]
+            result["from"] = network_from["address"]
+            result["family"] = family
+        if result["to"]:
+            network_to = result["to"]
+            family = network_to["family"]
+            result["to"] = network_to["address"]
+            result["family"] = family
+
+        if result["action"] == "to-table":
+            if result["table"] is None:
+                raise ValidationError(
+                    name,
+                    "missing table for the routing rule",
+                )
+
+        if result["from_len"] == 0:
+            if result["from"] is not None:
+                raise ValidationError(
+                    name,
+                    "the prefix length for from/src cannot be zero when from/src is specified",
+                )
+        elif result["from"] is None:
+            raise ValidationError(name, "missing from/src for a non zero prefix-length")
+        elif result["from_len"] > (32, 128)[result["family"] == socket.AF_INET6]:
```

Comment:
is this `[a, b](boolean_expression)` really a desirable pattern? Seems not good to me. It's hard to read, and does it actually require python to create a list? Maybe python can optimize this away, but it's not clear that it can.

Instead, add a function `Util.addr_family_prefix_length(prefix)`, and also use it from `Util.addr_family_valid_prefix()`.